### PR TITLE
Subscribe to Offer Sign Status

### PR DESCRIPTION
**Pull request by hedvigoscar**
I may have messed up the branch name a bit. But that's ok 🙂

This branch is based on `feature/sign-offer-mutation` (#9) - review that before this 🙏

### CODE REVIEW

File: src/api/index.ts [hunk 1]
```diff
@@ -81,7 +120,38 @@ const createProduct = (baseUrl: string, headers: ForwardHeaders) => async (
     },
     body: JSON.stringify(body),
   })
+  checkStatus(data)
```
Comment (palmfjord) [inline src/api/index.ts]: i really don't like doing these types of checks every time we want to call the api. Can't we just abstract the fetch into something we can reuse?

Comment (palmfjord) [reply src/api/index.ts]: woops, i'll leave the same comment in #9 hehe

File: src/api/index.ts [hunk 2]
```diff
@@ -57,92 +58,131 @@ interface CreateProductDto {
...
+  },
+) => Promise<Response>
+
+const callApi: CallApi = async (url, options = {}) => {
+  const { mergeOptions, token, validateStatus = checkStatus } = options
+  const headers: { [key: string]: string } = {
+    Accept: 'application/json',
+  }
+  if (mergeOptions!.method === 'POST') {
+    headers['Content-Type'] = 'application/json'
```
Comment (palmfjord) [inline src/api/index.ts]: i really don't like these kind of mutations on an object, but lets leave it for now, its quite isolated

File: src/api/index.ts [hunk 3]
```diff
@@ -57,92 +58,131 @@ interface CreateProductDto {
...
-
-const register = (baseUrl: string, headers: ForwardHeaders) => async () => {
-  const data = await fetch(`${baseUrl}/helloHedvig`, {
-    method: 'POST',
-    headers: headers as any,
+const register = async (headers: ForwardHeaders) => {
+  const data = await callApi('/helloHedvig', {
+    mergeOptions: {
+      method: 'POST',
+      headers: headers as any,
```
Comment (palmfjord) [inline src/api/index.ts]: why u no

typecast properly

File: src/api/index.ts (hunk 2 above)
Comment (hedvigoscar) [reply src/api/index.ts]: I agree actually, this is just the least bad way I've found to represent this sort of thing 😕 I'm open for suggestions if you have alternative ways to represent it!

**Pull request merged**